lib/html/pipeline/emoji_filter.rb

```diff
@@ -71,7 +71,42 @@ def asset_path(name)
 
       # Build an emoji image tag
       def emoji_image_tag(name)
-        "<img class='emoji' title=':#{name}:' alt=':#{name}:' src='#{emoji_url(name)}' height='20' width='20' align='absmiddle' />"
+        "<img #{img_html_attrs(name)}>"
+      end
+
+      def img_html_attrs(name)
+        img_attrs(name).map { |attr, value| %(#{attr}="#{value}") }.join(" ")
+      end
+
+      def img_attrs(name)
+        user_overrides = customized_attrs(name).select { |k, v| !v.nil? }
+        excluded_keys = customized_attrs(name).select { |k, v| v.nil? }.keys
+        result = default_img_attrs(name).merge!(user_overrides)
+        result.except(*excluded_keys)
+      end
```

Comment:
Could we simplify this to just merge the hashes and iterate through the values inline? This method creates several intermediate structures that aren't necessary and increase memory usage.